### DISCUSSION TITLE
Any SpecFlow / Reqnroll users know how to get the Cucumber extension to read binaries?

### DISCUSSION THREAD

**Discussion by btvanhooser** [cucumber/common#2193] [Visual Studio Code]
I'm unfamiliar with the usages in other languages with Cucumber directly, but SpecFlow (now Reqnroll) has a specflow.json/reqnroll.json file that allows you to specify binaries that your steps are located at in addition to your current binary. It seems that the Cucumber VS Code extension expects your steps to be all in your current workspace. We leaned really hard into NuGet package pattern of storing shared steps that are helpful across multiple testing solutions and now we have a new member using a MacBook without access to Rider. That leaves VS Code, but almost half of the steps on the feature files are marked as undefined because they are from the shared NuGet package steps. The tests run fine of course, but it's going to be really hard to on-board them with this setup. We can "un-roll" our shared steps pattern, but that's going to lead to maintenance concerns. Any recommendations? I tried diving into the "glue" setting but I could only see patterns for declaring files and not packages. I'm not familiar enough with Cucumber as a direct product to know if this is a ridiculous request, or if this is something other languages take advantage of or not (like with NPM, Maven, gemfiles, etc)

**Comment by mpkorstanje** [comment]
I'm not too familiar with .Net or the VSCode plugin so take this answer for what it is. 

To wit, the Cucumber VSCode plugin uses tree-sitter to find the step definitions in a source files. As .Net packages are distributed as binaries, I don't think tree sitter could scan these. But I also don't know what other means are available to discover step definitions for .Net that could work off dependencies.

If anything you may want to file a feature request for the VS Code plugin and may also want to ask at  https://github.com/orgs/reqnroll/discussions. Though given that building and maintaining IDE plugins is fairly complicated and time consuming I would not expect too much.

**Reply by btvanhooser** [reply]
Definitely appreciate the advice. I hesitated to ask the maintainers of reqnroll because it was their docs that recommend using this extension in the first place so I kinda took that as their answer. Maybe they'd still be willing to contribute this type of functionality on this plugin? Could be a nice compromise? I'll still make the feature request here to open the topic up. Thanks!

**Reply by mpkorstanje** [reply]
I'm not sure, but are you perhaps confusing https://github.com/reqnroll/Reqnroll.VisualStudio and https://github.com/cucumber/vscode?

**Reply by btvanhooser** [reply]
Nope! I'm aware of the of the VS extension. That's what we use now, but the new member would be using a mac book and VS just got deprecated there so they won't have Visual Studio (https://visualstudio.microsoft.com/vs/mac/). Reqnroll actually calls out this project's extension by link: https://docs.reqnroll.net/latest/installation/setup-ide.html#setup-vscode

I'll get the discussion going somewhere for sure since they do need a solution in some way.

**Reply by richard-haslam** [reply]
@btvanhooser did you ever find a solution for this? I am currently trying to solve the same problem and so far none of the workarounds I can find are quite right.

**Reply by btvanhooser** [reply]
> @btvanhooser did you ever find a solution for this? I am currently trying to solve the same problem and so far none of the workarounds I can find are quite right.

@richard-haslam nope! I just lucked out and the new member didn't end up landing on our team so we didn't need to prioritize this. It's still very much so desired, but the context you see here is as far as we got.
